packages/google_sign_in/google_sign_in_web/lib/src/gis_client.dart

```diff
@@ -302,9 +321,17 @@ class GisSdkClient {
     await signOut();
   }
 
-  /// Returns true if the client has recognized this user before.
+  /// Returns true if the client has recognized this user before, and the last-seen
+  /// credential is not expired.
   Future<bool> isSignedIn() async {
-    return _lastCredentialResponse != null || _requestedUserData != null;
+    bool isSignedIn = false;
+    if (_lastCredentialResponse != null) {
+      final DateTime? expiration = utils
+          .getCredentialResponseExpirationTimestamp(_lastCredentialResponse);
+      isSignedIn = expiration?.isAfter(DateTime.now()) ?? false;
```

Comment:
So a missing `expiration` field would indicate that it's already expired? Naively (not knowing the API surface) I would expect that if expiration is optional, then not having it should yield a `true` result here, so maybe a comment would be helpful?

Reply:
I'm going to document this a little bit further, but in this case, if the `CredentialResponse` payload cannot be decoded, is null, or doesn't contain an "exp" claim, we assume that the token is bad, and thus the user is *not* authenticated.

In a valid Google ID Token, the "exp" claim should **always** be provided:

https://developers.google.com/identity/openid-connect/openid-connect#an-id-tokens-payload